### ISSUE TITLE
SSLV3 HANDSHAKE FAILURE

### ISSUE THREAD

**Issue by moty66** [foospidy/GrepBugs#5]
Logfile:

<pre>
2015-10-12 14:59:45 INFO Using grep binary grep
2015-10-12 14:59:45 INFO Starting local scan with scan id 1cd8ed0c-70e1-11e5-9e79-10ddb1b3b89f
2015-10-12 14:59:45 INFO Retreiving rules from http://grepbugs.com/json
2015-10-12 14:59:45 INFO Starting new HTTP connection (1): grepbugs.com
2015-10-12 14:59:45 DEBUG "GET /json HTTP/1.1" 301 None
2015-10-12 14:59:45 INFO Starting new HTTPS connection (1): grepbugs.com
2015-10-12 14:59:46 WARNING Error retreiving grep rules: ConnectionError (attempt 1 of 3): [SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:590)
2015-10-12 14:59:49 INFO Starting new HTTP connection (1): grepbugs.com
2015-10-12 14:59:49 DEBUG "GET /json HTTP/1.1" 301 None
2015-10-12 14:59:49 INFO Starting new HTTPS connection (1): grepbugs.com
2015-10-12 14:59:49 WARNING Error retreiving grep rules: ConnectionError (attempt 2 of 3): [SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:590)
2015-10-12 14:59:52 INFO Starting new HTTP connection (1): grepbugs.com
2015-10-12 14:59:52 DEBUG "GET /json HTTP/1.1" 301 None
2015-10-12 14:59:52 INFO Starting new HTTPS connection (1): grepbugs.com
2015-10-12 14:59:52 WARNING Error retreiving grep rules: ConnectionError (attempt 3 of 3): [SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:590)
2015-10-12 14:59:55 DEBUG Error retreiving grep rules (no more tries left. could be using old grep rules.): [SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:590)
2015-10-12 14:59:55 INFO Running cloc for sql output.
2015-10-12 15:01:35 INFO Reading grep rules from json file.
2015-10-12 15:01:35 CRITICAL Unhandled exception: [Errno 2] No such file or directory: '/Users/xxxx/Desktop/grepbugs/data/grepbugs.json'
</pre>

uname -a

<pre>
Darwin xxxxxx.local 14.5.0 Darwin Kernel Version 14.5.0: Wed Jul 29 02:26:53 PDT 2015; root:xnu-2782.40.9~1/RELEASE_X86_64 x86_64
</pre>

other details

<pre>
OpenSSL 1.0.2d 9 Jul 2015
cloc 1.64
Python 2.7.10
</pre>

**Comment by foospidy** [comment]
i'm getting the same error on osx... looking into it.

**Comment by foospidy** [comment]
I don't have a great answer. I tried updating various python libraries (https://stackoverflow.com/questions/18578439/using-requests-with-tls-doesnt-give-sni-support) and that didn't work. I also tried the options listed here: https://github.com/kennethreitz/requests/issues/2022#issuecomment-41475570 still didn't work. The best I can suggest right now is to cd to the GrepBugs/data directory and run: wget https://grepbugs.com/rules -o grepbugs.json

When you run grepbugs the SSL handshake will still fail, but it will continue with the local  grepbugs.json file. Perhaps I should add using the wget command as a fallback method in grepbugs.py
